mergify_engine/rules/__init__.py

```diff
@@ -454,6 +577,26 @@ def YAML(v: bytes) -> typing.Any:
         raise YAMLInvalid(message="Invalid YAML", error_message=error_message)
 
 
+def RuleConditionSchema(v: typing.Any, depth: int = 0) -> typing.Any:
+    if depth > 3:
+        raise voluptuous.Invalid("maximun number of nested and/or reached")
```

Comment:
This does not seem documented.
```suggestion
    if depth > 3:
        raise voluptuous.Invalid("Maximum number of nested conditions reached")
```